include/aws/common/byte_buf.h

```diff
@@ -438,85 +473,103 @@ static inline bool aws_byte_cursor_read_be64(struct aws_byte_cursor *cur, uint64
 }
 
 /**
- * Write specified number of bytes from array to byte cursor.
+ * Write specified number of bytes from array to byte buffer.
  *
- * On success, returns true and updates the cursor pointer/length accordingly.
- * If there is insufficient space in the cursor, returns false, leaving the
- * cursor unchanged.
+ * On success, returns true and updates the buffer length accordingly.
+ * If there is insufficient space in the buffer, returns false, leaving the
+ * buffer unchanged.
  */
-static inline bool aws_byte_cursor_write(
-    struct aws_byte_cursor *AWS_RESTRICT cur,
+static inline bool aws_byte_buf_write(
+    struct aws_byte_buf *AWS_RESTRICT cur,
     const uint8_t *AWS_RESTRICT src,
     size_t len) {
-    struct aws_byte_cursor slice = aws_byte_cursor_advance_nospec(cur, len);
+
+    struct aws_byte_cursor write_cursor;
+    write_cursor.ptr = cur->buffer + cur->len;
+    write_cursor.len = cur->capacity - cur->len;
+    struct aws_byte_cursor slice = aws_byte_cursor_advance(&write_cursor, len);
```

Comment:
Why are we doing this dance with cursors here?

Reply:
Originally I had this using the `nospec` version and didn't want to rewrite it for buffers, but I should be able to factor it out now.